### PR TITLE
refactor(rxIfEnvironment): EO-1151 Refactor rxIfEnvironment

### PR DESCRIPTION
**Pull request by satyamkotakonda**
JIRA:  https://jira.rax.io/browse/EO-1151

### LGTMs
- [x] Dev LGTM
- [x] Dev+Vidyo LGTM
- [x] Design LGTM
- ~~[ ] QE LGTM (no change to functional JavaScript)~~

Before
--------
screen shot 2016-09-16 at 1 51 31 pm

After
-----

screen shot 2016-09-21 at 4 36 25 pm

screen shot 2016-09-21 at 4 32 00 pm

### CODE REVIEW

File: src/utilities/rxEnvironment/rxEnvironment.meta.json [hunk 1]
```diff
@@ -0,0 +1,11 @@
+{
+    "displayName": "rxEnvironment",
+    "stability": "stable",
+    "description": "",
+    "api": "directive:rxEnvironment",
+    "keywords": [
+      "rxEnvironment",
```
Comment (khaziwallis) [inline src/utilities/rxEnvironment/rxEnvironment.meta.json]: @satyamkotakonda  4 space please

**Comment by satyamkotakonda** [comment]
@khaziwallis  Updated changes and please review.

File: src/utilities/rxEnvironment/docs/rxEnvironment.docs.html [hunk 2]
```diff
@@ -0,0 +1,6 @@
+<p>
+  Utilities built to detect and provide the current environment (e.g. dev, staging, prod)
```
Comment (KalpeshBhandari) [inline src/utilities/rxEnvironment/docs/rxEnvironment.docs.html]: Keep singular "Utility"

File: src/components/rxEnvironment/rxEnvironment.module.js [hunk 3]
```diff
@@ -1,69 +0,0 @@
...
- * @description
- * # rxEnvironment Component
- *
- * Component built to detect and provide the current environment (e.g. dev, staging, prod)
- *
- * ## Current Environments ##
- *
- * This service defines the following Encore specific environments:
- *
- * * **local** - http://localhost:port and http://server:port
```
Comment (KalpeshBhandari) [inline src/components/rxEnvironment/rxEnvironment.module.js]: Are we removing these ngdoc sections?

File: src/utilities/rxEnvironment/docs/examples/rxEnvironment.simple.html [hunk 4]
```diff
@@ -0,0 +1,19 @@
+<div ng-controller="rxEnvironmentSimpleCtrl">
+  <p>Url built from Environment data: {{ { tld: 'cloudatlas', path: 'some/path' } | rxEnvironmentUrl }}</p>
```
Comment (KalpeshBhandari) [inline src/utilities/rxEnvironment/docs/examples/rxEnvironment.simple.html]: Para content is inconsistent at lines 2, 4, 10, and 16.
I guess line 16 format looks good, and just to be consistent apply it to other above instances.

File: src/utilities/rxEnvironment/scripts/rxIfEnvironment.js [hunk 5]
```diff
@@ -1,7 +1,7 @@
-angular.module('encore.ui.rxEnvironment')
+angular.module('encore.ui.utilities')
 /**
  * @ngdoc directive
- * @name rxEnvironment.directive:rxEnvironment
+ * @name encore.ui.utilities.directive:rxEnvironment
```
Comment (KalpeshBhandari) [inline src/utilities/rxEnvironment/scripts/rxIfEnvironment.js]: "View API" link is broken

Comment (CITguy) [inline src/utilities/rxEnvironment/scripts/rxIfEnvironment.js]: This is fine.

File: src/components/rxApp/scripts/rxAppNav.js [hunk 6]
```diff
@@ -263,7 +263,7 @@ angular.module('encore.ui.rxApp')
  *
  * This expression would be evaluated, checking if the user is currently viewing
  * the app in the `unified-preprod` environment or the `local` environment, and
- * only display the item if one of those was true. (See {@link rxEnvironment}
+ * only display the item if one of those was true. (See {@link encore.ui.utilities.directive:rxEnvironment}
```
Comment (CITguy) [inline src/components/rxApp/scripts/rxAppNav.js]: Link should point to `utilities.service:Environment`

File: src/utilities/rxEnvironment/scripts/rxIfEnvironment.js (hunk 5 above)
Comment (CITguy) [inline src/utilities/rxEnvironment/scripts/rxIfEnvironment.js]: This should be `@name utilities.directive:rxIfEnvironment`

File: src/components/rxEnvironment/rxEnvironment.module.js (hunk 3 above)
Comment (CITguy) [inline src/components/rxEnvironment/rxEnvironment.module.js]: Please migrate the ngDocs to `src/utilities/Environment/scripts/Environment.js`

File: src/components/rxEnvironment/rxEnvironment.module.js [hunk 7]
```diff
@@ -1,69 +0,0 @@
...
- * the name of the current environment. This pattern should be avoided,
- * specifically because of the overlapping environment issue discussed above.
- * If you call `Environment.get().name`, it will just return the first matching
- * environment in the list of environments, even if we're overlapping and have
- * multiple environments. Instead, check explicitly with
- * `Environment.isLocal()`, `Environment.isPreProd()`, etc., or
- * use `Environment.envCheck('local')`
- *
- * ## Directives
- * * {@link rxEnvironment.directive:rxEnvironment rxEnvironment}
```
Comment (CITguy) [inline src/components/rxEnvironment/rxEnvironment.module.js]: Do not migrate "Directives" section of ngDocs.

File: src/utilities/rxEnvironment/docs/examples/rxEnvironment.simple.html [hunk 8]
```diff
@@ -0,0 +1,19 @@
...
+  <p>Content shows if not on GitHub demo page:
+    <span rx-if-environment="!ghPages">
+      You're not on the GitHub demo page
+    </span>
+  </p>
+
+  <p>
+    Are we in the <code>ghPages</code> environment? {{ 'ghPages' | rxEnvironmentMatch }}
+  </p>
+</div>
```
Comment (CITguy) [inline src/utilities/rxEnvironment/docs/examples/rxEnvironment.simple.html]: File should be moved to `src/utilities/rxIfEnvironment/docs/examples/rxIfEnvironment.simple.html`

File: src/utilities/rxEnvironment/docs/examples/rxEnvironment.simple.js [hunk 9]
```diff
@@ -0,0 +1,4 @@
+angular.module('demoApp')
+.controller('rxEnvironmentSimpleCtrl', function ($scope, Environment) {
```
Comment (CITguy) [inline src/utilities/rxEnvironment/docs/examples/rxEnvironment.simple.js]: Controller should be named `rxIfEnvironmentSimpleCtrl`

Comment (CITguy) [inline src/utilities/rxEnvironment/docs/examples/rxEnvironment.simple.js]: File should be moved to `src/utilities/rxIfEnvironment/docs/examples/rxIfEnvironment.simple.js`

File: src/utilities/rxEnvironment/docs/rxEnvironment.docs.html (hunk 2 above)
Comment (CITguy) [inline src/utilities/rxEnvironment/docs/rxEnvironment.docs.html]: File should be moved to `src/utilities/rxIfEnvironment/docs/rxIfEnvironment.docs.html`

File: src/utilities/rxEnvironment/rxEnvironment.meta.json [hunk 10]
```diff
@@ -0,0 +1,11 @@
+{
+    "displayName": "rxEnvironment",
```
Comment (CITguy) [inline src/utilities/rxEnvironment/rxEnvironment.meta.json]: Display name should be `rxIfEnvironment`

Comment (CITguy) [inline src/utilities/rxEnvironment/rxEnvironment.meta.json]: File should be moved to `src/utilities/rxIfEnvironment/rxIfEnvironment.meta.json`

File: src/utilities/rxEnvironment/scripts/rxEnvironment.spec.js [hunk 11]
```diff
@@ -3,7 +3,7 @@ describe('rxEnvironment', function () {
 
     beforeEach(function () {
         // load module
```
Comment (CITguy) [inline src/utilities/rxEnvironment/scripts/rxEnvironment.spec.js]: File should be moved to `src/utilities/Environment/scripts/Environment.spec.js`

Comment (CITguy) [inline src/utilities/rxEnvironment/scripts/rxEnvironment.spec.js]: Line 1 should have description of `utilities:Environment`

File: src/utilities/rxEnvironment/scripts/rxIfEnvironment.js (hunk 5 above)
Comment (CITguy) [inline src/utilities/rxEnvironment/scripts/rxIfEnvironment.js]: File should be moved to `src/utilities/rxIfEnvironment/scripts/rxIfEnvironment.js`

File: src/utilities/rxEnvironment/scripts/rxIfEnvironment.spec.js [hunk 12]
```diff
@@ -7,7 +7,7 @@ describe('rxIfEnvironment', function () {
 
     beforeEach(function () {
```
Comment (CITguy) [inline src/utilities/rxEnvironment/scripts/rxIfEnvironment.spec.js]: File should be moved to `src/utilities/rxIfEnvironment/scripts/rxIfEnvironment.spec.js`

File: utils/rx-page-objects/test/rxEnvironment.midway.js [hunk 13]
```diff
@@ -3,7 +3,7 @@ var environment = encore.rxEnvironment;
 describe('rxEnvironment', function () {
```
Comment (CITguy) [inline utils/rx-page-objects/test/rxEnvironment.midway.js]: Description should be `rxIfEnvironment`

Comment (CITguy) [inline utils/rx-page-objects/test/rxEnvironment.midway.js]: File should be moved to `utils/rx-page-objects/test/rxIfEnvironment.midway.js`

File: utils/rx-page-objects/test/rxEnvironment.midway.js [hunk 14]
```diff
@@ -3,7 +3,7 @@ var environment = encore.rxEnvironment;
...
 
     before(function () {
-        demoPage.go('#/components/rxEnvironment');
+        demoPage.go('#/utilities/rxEnvironment');
```
Comment (CITguy) [inline utils/rx-page-objects/test/rxEnvironment.midway.js]: should be `#/utilities/rxIfEnvironment`

File: src/utilities/rxEnvironment/rxEnvironment.meta.json [hunk 15]
```diff
@@ -0,0 +1,11 @@
...
+    "stability": "stable",
+    "description": "",
+    "api": "directive:rxEnvironment",
```
Comment (CITguy) [inline src/utilities/rxEnvironment/rxEnvironment.meta.json]: should be `directive:rxIfEnvironment`

File: src/utilities/rxEnvironment/rxEnvironment.meta.json [hunk 16]
```diff
@@ -0,0 +1,11 @@
...
+    "keywords": [
+        "detect",
+        "environment",
+        "rxEnvironment"
```
Comment (CITguy) [inline src/utilities/rxEnvironment/rxEnvironment.meta.json]: please change keyword to `rxIfEnvironment`

File: src/utilities/rxEnvironment/scripts/rxEnvironment.spec.js (hunk 11 above)
Comment (satyamkotakonda) [inline src/utilities/rxEnvironment/scripts/rxEnvironment.spec.js]: @CITguy  Can you please provide the description for this.

File: src/utilities/rxEnvironment/rxEnvironment.meta.json (hunk 16 above)
Comment (satyamkotakonda) [inline src/utilities/rxEnvironment/rxEnvironment.meta.json]: @CITguy  

 Can you please elaborate this. 

```
 Looks like there was some confusion with naming.

The old rxEnvironment component housed documentation and tests for both the Environment service as well as rxIfEnvironment directive. Please move files accordingly.

**Comment by satyamkotakonda** [comment]
@CITguy  Updated changes and please review.
Thanks.

File: src/utilities/rxIfEnvironment/docs/rxIfEnvironment.docs.html [hunk 17]
```diff
@@ -0,0 +1,6 @@
+<p>
+  Utility built to detect and provide the current environment (e.g. dev, staging, prod)
+</p>
+
+<h3><rx-permalink>rxEnvironment</rx-permalink></h3>
+<rx-example name="rxEnvironment.simple"></rx-example>
```
Comment (CITguy) [inline src/utilities/rxIfEnvironment/docs/rxIfEnvironment.docs.html]: Please change name to `rxIfEnvironment.simple`. This errors when trying to view the demo page.

File: src/utilities/rxIfEnvironment/docs/examples/rxIfEnvironment.simple.html [hunk 18]
```diff
@@ -0,0 +1,23 @@
+<div ng-controller="rxIfEnvironmentSimpleCtrl">
+  <p>
```
Comment (CITguy) [inline src/utilities/rxIfEnvironment/docs/examples/rxIfEnvironment.simple.html]: Paragraph is unnecessary. This is already handled by the `rxEnvironmentUrl` utility demo.

Please remove Lines 2-5.

File: src/utilities/rxIfEnvironment/docs/examples/rxIfEnvironment.simple.html [hunk 19]
```diff
@@ -0,0 +1,23 @@
...
+    Content shows if not on GitHub demo page:
+    <span rx-if-environment="!ghPages">
+      You're not on the GitHub demo page
+    </span>
+  </p>
+
+  <p>
+    Are we in the <code>ghPages</code> environment? {{ 'ghPages' | rxEnvironmentMatch }}
+  </p>
```
Comment (CITguy) [inline src/utilities/rxIfEnvironment/docs/examples/rxIfEnvironment.simple.html]: Paragraph is unnecessary. This is already handled by the `rxEnvironmentMatch` utility demo.

Please remove lines 19-22.

File: src/utilities/rxIfEnvironment/docs/examples/rxIfEnvironment.simple.js [hunk 20]
```diff
@@ -0,0 +1,4 @@
+angular.module('demoApp')
+.controller('rxIfEnvironmentSimpleCtrl', function ($scope, Environment) {
+    $scope.Environment = Environment;
+});
```
Comment (CITguy) [inline src/utilities/rxIfEnvironment/docs/examples/rxIfEnvironment.simple.js]: Controller is unnecessary. Please remove file.

File: src/utilities/rxIfEnvironment/docs/examples/rxIfEnvironment.simple.html [hunk 21]
```diff
@@ -0,0 +1,23 @@
+<div ng-controller="rxIfEnvironmentSimpleCtrl">
```
Comment (CITguy) [inline src/utilities/rxIfEnvironment/docs/examples/rxIfEnvironment.simple.html]: `<div>` is unnecessary. Keep content, but remove wrapping `<div>`.

File: src/utilities/rxEnvironment/rxEnvironment.meta.json (hunk 16 above)
Comment (CITguy) [inline src/utilities/rxEnvironment/rxEnvironment.meta.json]: That was purely a comment noting that `rxEnvironment` acted as a namespace that housed `Environment` and `rxIfEnvironment` stuff.

What you've updated is a move toward the desired outcome.

File: src/utilities/Environment/scripts/Environment.js [hunk 22]
```diff
@@ -41,6 +41,64 @@ angular.module('encore.ui.utilities')
  * <pre>
  * Environment.get() // return environment object that matches current location
  * </pre>
+ *
+
+ * Component built to detect and provide the current environment (e.g. dev, staging, prod)
```
Comment (CITguy) [inline src/utilities/Environment/scripts/Environment.js]: Please move this documentation under the `@description` (above `@example` on line 40)

**Comment by satyamkotakonda** [comment]
@CITguy  Addressed review comments and please review.

File: src/utilities/rxIfEnvironment/docs/examples/rxIfEnvironment.simple.html [hunk 23]
```diff
@@ -0,0 +1,13 @@
+<p>
+  Content shows if on GitHub demo page:
+  <span rx-if-environment="ghPages">
+      You're on the GitHub demo page
+    </span>
```
Comment (CITguy) [inline src/utilities/rxIfEnvironment/docs/examples/rxIfEnvironment.simple.html]: Please correct indentation.

File: src/utilities/rxIfEnvironment/docs/examples/rxIfEnvironment.simple.html [hunk 24]
```diff
@@ -0,0 +1,13 @@
...
+</p>
+
+<p>
+  Content shows if not on GitHub demo page:
+  <span rx-if-environment="!ghPages">
+      You're not on the GitHub demo page
+    </span>
```
Comment (CITguy) [inline src/utilities/rxIfEnvironment/docs/examples/rxIfEnvironment.simple.html]: Please correct indentation.

**Comment by satyamkotakonda** [comment]
@CITguy  Indentation issue fixed and sorry for that I have missed because of shortcuts usage.

**Comment by KalpeshBhandari** [comment]
Dev LGTM

**Comment by cgtaab** [comment]
Design LGTM

**Pull request merged**